src/compile.rs

```diff
@@ -133,10 +133,27 @@ fn compile_target(
         }
     }
 
+    let module_name = &context.module_name;
+    let so_filename = match python_interpreter {
+        Some(python_interpreter) => python_interpreter.get_library_name(module_name),
+        // abi3
+        None => {
+            format!("{base}.abi3.so", base = module_name)
+        }
+    };
+    let macos_dylib_install_name = format!("link-args=-Wl,-install_name,@rpath/{}", so_filename);
```

Comment:
Please add a comment what this does / why this is required

Reply:
How about a link to https://github.com/PyO3/setuptools-rust/issues/106 ?

Reply:
Yep, that should do